src/jv_aux.c

```diff
@@ -151,13 +151,16 @@ jv jv_get(jv t, jv k) {
     jv_free(t);
     jv_free(k);
   }
+
+  if(jv_is_borrowed(v)) return jv_unborrow(v);
   return v;
 }
 
 jv jv_set(jv t, jv k, jv v) {
   if (!jv_is_valid(v)) {
     jv_free(t);
     jv_free(k);
+    if(jv_is_borrowed(v)) return jv_unborrow(v);
```

Comment:
Hmmm, before this PR `v` here would be an owned or copied invalid value, and we would return... the same thing, without incrementing or decrementing the reference count on `v`.  We should do the same here, but indeed, if `v` is borrowed we need to return it not-borrowed, but... _with or without incrementing the reference count_?

Reply:
We absolutely need to increment the reference count else the following could and has occurred(this caused me quite some pain)

```c
jv a = jv_string("to borrow");
jv b = JV_ARRAY(jv_borrow(a));
jv_free(a);

jv c = jv_getpath(b, JV_ARRAY(jv_number(0))); // trying to read deleted memory
```